dbms/src/TestUtils/executorSerializer.cpp

```diff
@@ -0,0 +1,265 @@
+    // currently only support "and" function in selection executor.
+    buf.joinStr(
+        sel.conditions().begin(),
+        sel.conditions().end(),
+        [&](const auto & expr, FmtBuffer &) {
+            serializeExpression(expr, buf);
+        },
+        " and ");
+    buf.append("}\n");
+}
+
+void serializeLimit(const String & executor_id, const tipb::Limit & limit, FmtBuffer & buf)
+{
+    buf.fmtAppend("{} | {}\n", executor_id, limit.limit());
+}
+
+void serializeProjection(const String & executor_id, const tipb::Projection & proj, FmtBuffer & buf)
+{
+    buf.fmtAppend("{} | {{", executor_id);
+    buf.joinStr(
+        proj.exprs().begin(),
+        proj.exprs().end(),
+        [&](const auto & expr, FmtBuffer &) {
+            serializeExpression(expr, buf);
+        },
+        ", ");
+    buf.append("}\n");
+}
+
+void serializeAggregation(const String & executor_id, const tipb::Aggregation & agg, FmtBuffer & buf)
+{
+    buf.fmtAppend("{} | group_by: {{", executor_id);
+    buf.joinStr(
+        agg.group_by().begin(),
+        agg.group_by().end(),
+        [&](const auto & group_by, FmtBuffer &) {
+            serializeExpression(group_by, buf);
+        },
+        ", ");
+    buf.append("}, agg_func: {");
+    buf.joinStr(
+        agg.agg_func().begin(),
+        agg.agg_func().end(),
+        [&](const auto & func, FmtBuffer &) {
+            serializeExpression(func, buf);
+        },
+        ", ");
+    buf.append("}\n");
+}
+
+void serializeTopN(const String & executor_id, const tipb::TopN & top_n, FmtBuffer & buf)
+{
+    buf.fmtAppend("{} | order_by: {{", executor_id);
+    buf.joinStr(
+        top_n.order_by().begin(),
+        top_n.order_by().end(),
+        [&](const auto & order_by, FmtBuffer & fb) {
+            fb.append("(");
+            serializeExpression(order_by.expr(), buf);
+            fb.fmtAppend(", desc: {})", order_by.desc());
+        },
+        ", ");
+    buf.fmtAppend("}}, limit: {}\n", top_n.limit());
+}
+
+void serializeJoin(const String & executor_id, const tipb::Join & join, FmtBuffer & buf)
+{
+    assert(join.left_join_keys_size() > 0);
+    assert(join.right_join_keys_size() > 0);
+    buf.fmtAppend("{} | {}, {}. left_join_keys: {{", executor_id, getJoinTypeName(join.join_type()), getJoinExecTypeName(join.join_exec_type()));
+    int bound = join.left_join_keys_size() - 1;
+    for (int i = 0; i < bound; ++i)
+    {
+        buf.fmtAppend("<{}, {}>, ", i, getFieldTypeName(join.left_join_keys(i).field_type().tp()));
+    }
+    buf.fmtAppend("<{}, {}>", bound, getFieldTypeName(join.left_join_keys(bound).field_type().tp()));
+    buf.append("}, right_join_keys: {");
+    bound = join.right_join_keys_size() - 1;
+    for (int i = 0; i < bound; ++i)
+    {
+        buf.fmtAppend("<{}, {}>, ", i, getFieldTypeName(join.right_join_keys(i).field_type().tp()));
+    }
+    buf.fmtAppend("<{}, {}>", bound, getFieldTypeName(join.right_join_keys(bound).field_type().tp()));
```

Comment:
how about wrap a  function in `executorSerizlize.cpp`

Reply:
Good idea! I will fix it.